From ef7aeba83a6fb936c4dca847a69140c5b22c8000 Mon Sep 17 00:00:00 2001
From: sebres <serg.brester@sebres.de>
Date: Fri, 7 Feb 2020 11:08:01 +0100
Subject: [PATCH] amend to f3dbc9dda10e52610e3de26f538b5581fd905505: change
 main thread-name back to `fail2ban-server`; implements new command line
 option `--pname` to specify it by start of server (default
 `fail2ban-server`); closes gh-2623 (revert change of main thread-name,
 because it can affect process-name too, so `pgrep` & co. may be confused)

(cherry picked from commit 3f489070646b363aa0374681fe910f05521cd247)
---
 fail2ban/client/fail2bancmdline.py | 3 ++-
 fail2ban/server/server.py          | 5 +++--
 man/fail2ban-client.1              | 3 +++
 man/fail2ban-server.1              | 3 +++
 4 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/fail2ban/client/fail2bancmdline.py b/fail2ban/client/fail2bancmdline.py
index 1268ee9f..3c6bd0bf 100644
--- a/fail2ban/client/fail2bancmdline.py
+++ b/fail2ban/client/fail2bancmdline.py
@@ -97,6 +97,7 @@ class Fail2banCmdLine():
 		output("    -c <DIR>                configuration directory")
 		output("    -s <FILE>               socket path")
 		output("    -p <FILE>               pidfile path")
+		output("    --pname <NAME>          name of the process (main thread) to identify instance (default fail2ban-server)")
 		output("    --loglevel <LEVEL>      logging level")
 		output("    --logtarget <TARGET>    logging target, use file-name or stdout, stderr, syslog or sysout.")
 		output("    --syslogsocket auto|<FILE>")
@@ -185,7 +186,7 @@ class Fail2banCmdLine():
 			try:
 				cmdOpts = 'hc:s:p:xfbdtviqV'
 				cmdLongOpts = ['loglevel=', 'logtarget=', 'syslogsocket=', 'test', 'async',
-					'timeout=', 'str2sec=', 'help', 'version', 'dp', '--dump-pretty']
+					'pname=', 'timeout=', 'str2sec=', 'help', 'version', 'dp', '--dump-pretty']
 				optList, self._args = getopt.getopt(self._argv[1:], cmdOpts, cmdLongOpts)
 			except getopt.GetoptError:
 				self.dispUsage()
diff --git a/fail2ban/server/server.py b/fail2ban/server/server.py
index 15265822..417f4514 100644
--- a/fail2ban/server/server.py
+++ b/fail2ban/server/server.py
@@ -81,8 +81,6 @@ class Server:
 			'Linux': '/dev/log',
 		}
 		self.__prev_signals = {}
-		# replace real thread name with short process name (for top/ps/pstree or diagnostic):
-		prctl_set_th_name('f2b/server')
 
 	def __sigTERMhandler(self, signum, frame): # pragma: no cover - indirect tested
 		logSys.debug("Caught signal %d. Exiting", signum)
@@ -113,6 +111,9 @@ class Server:
 				logSys.error(err)
 				raise ServerInitializationError(err)
 			# We are daemon.
+
+		# replace main thread (and process) name to identify server (for top/ps/pstree or diagnostic):
+		prctl_set_th_name(conf.get("pname", "fail2ban-server"))
 		
 		# Set all logging parameters (or use default if not specified):
 		self.__verbose = conf.get("verbose", None)
diff --git a/man/fail2ban-client.1 b/man/fail2ban-client.1
index 745c080a..5727caf0 100644
--- a/man/fail2ban-client.1
+++ b/man/fail2ban-client.1
@@ -19,6 +19,9 @@ socket path
 \fB\-p\fR <FILE>
 pidfile path
 .TP
+\fB\-\-pname\fR <NAME>
+name of the process (main thread) to identify instance (default fail2ban\-server)
+.TP
 \fB\-\-loglevel\fR <LEVEL>
 logging level
 .TP
diff --git a/man/fail2ban-server.1 b/man/fail2ban-server.1
index 418b46dd..25c5ec96 100644
--- a/man/fail2ban-server.1
+++ b/man/fail2ban-server.1
@@ -19,6 +19,9 @@ socket path
 \fB\-p\fR <FILE>
 pidfile path
 .TP
+\fB\-\-pname\fR <NAME>
+name of the process (main thread) to identify instance (default fail2ban\-server)
+.TP
 \fB\-\-loglevel\fR <LEVEL>
 logging level
 .TP
-- 
2.20.1

